### PR TITLE
selinux: Remove copied checkmark when switching SELinux modifications tab

### LINKED ISSUE #19057: Switching selinux automation script tabs still shows text is copied (but it goes away after a while)
### Explain what happens

Copy the ansible script and then switch to the shell script version, notice that the clipboard still shows as copied.

[Uploading Screencast from 2023-07-04 09-32-17.webm…]()

### PR DESCRIPTION
**Pull request by omarY23**
Resolves #19057 .  Added support of clearTimeOut function to re-render the copy icon in case of switching tabs in selinux automation script.

### CODE REVIEW

**Comment by jelly** [comment]
While inspecting your PR I noticed your Cockpit fork is from March (basically quite old) I'd recommend syncing it to the latest `main`.

File: pkg/lib/cockpit-components-modifications.jsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ import cockpit from "cockpit";
 import 'cockpit-components-modifications.css';
 
 const _ = cockpit.gettext;
+tId = undefined;
```
Comment (jelly) [inline pkg/lib/cockpit-components-modifications.jsx:35]: this should be `let tId` be part of the component, we don't want to introduce new global variables.

File: pkg/lib/cockpit-components-modifications.jsx [hunk 2]
```diff
@@ -53,16 +54,18 @@ class ModificationsExportDialog extends React.Component {
 
     handleSelect(event, active_tab) {
         this.setState({ active_tab });
+        clearTimeout(this.tId);
```
Comment (jelly) [inline pkg/lib/cockpit-components-modifications.jsx:57]: This should also set `copied: false`, as that shows / toggles the green icon. The `clearTimeout` isn't actually needed but it just feels correct to do it. (Also it could introduce a race when a user clicks the button quickly after switching a tab).

**Review by jelly — CHANGES_REQUESTED** [review]

File: pkg/lib/cockpit-components-modifications.jsx [hunk 3]
```diff
@@ -53,18 +52,17 @@ class ModificationsExportDialog extends React.Component {
...
     copyToClipboard() {
         try {
             navigator.clipboard.writeText(this.props[this.state.active_tab].trim())
                     .then(() => {
                         this.setState({ copied: true });
-                        const timeOutId = setTimeout(() => {
+                        setTimeout(() => {
                             this.setState({ copied: false });
                         }, 3000);
-                        this.tId = timeOutId;
```
Comment (jelly) [inline pkg/lib/cockpit-components-modifications.jsx]: Cancelling the setTimeout with `clearTimeout` is still valid, in theory one could quickly switch tabs and click copy to clipboard and the previous set timeout would cancel it.

**Review by jelly — CHANGES_REQUESTED** [review]

File: pkg/lib/cockpit-components-modifications.jsx (hunk 3 above)
Comment (omarY23) [reply pkg/lib/cockpit-components-modifications.jsx]: Okay

File: pkg/storaged/content-views.jsx [hunk 4]
```diff
@@ -905,6 +905,7 @@ export class VGroup extends React.Component {
                     Message(cockpit.format(_("The $0 package will be installed to create VDO devices."), vdo_package),
                             {
                                 visible: vals => vals.purpose === 'vdo' && need_vdo_install,
+                                style: { marginTop: '1px' }
```
Comment (jelly) [inline pkg/storaged/content-views.jsx:908]: This change is unrelated and should not be in this PR.

**Review by jelly — CHANGES_REQUESTED** [review]

File: pkg/storaged/content-views.jsx (hunk 4 above)
Comment (omarY23) [reply pkg/storaged/content-views.jsx]: Ya i think its an old change and don't know how it got committed when I only added one file to commit.

**Review by jelly — APPROVED** [review]

File: pkg/lib/cockpit-components-modifications.jsx [hunk 5]
```diff
@@ -51,15 +52,16 @@ class ModificationsExportDialog extends React.Component {
     }
 
     handleSelect(event, active_tab) {
-        this.setState({ active_tab });
+        this.setState({ active_tab, copied: false });
+        clearTimeout(this.timeoutId);
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-modifications.jsx:56]: We usually call this as `window.clearTimeout`, and it should explicitly check if `timeoutId  !== null`. It *seems* to work with calling with null in current Firefox, but I wouldn't bet on it on all browsers.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Nice fix, thank you!

**Review by martinpitt — APPROVED** [review]
Thanks!

File: pkg/lib/cockpit-components-modifications.jsx [hunk 6]
```diff
@@ -51,15 +52,19 @@ class ModificationsExportDialog extends React.Component {
     }
 
     handleSelect(event, active_tab) {
-        this.setState({ active_tab });
+        this.setState({ active_tab, copied: false });
+        if (this.timeoutId !== null) {
+            clearTimeout(this.timeoutId);
+            this.timeoutId = null;
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-modifications.jsx:58]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19069-20230831-104711-4f88c3d0-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)
